### PR TITLE
Optimize route creation

### PR DESCRIPTION
**Pull request by itkg-nanne**
[OO-MISC] optimize route generation on node update

### CODE REVIEW

File: src/OpenOrchestra/Tests/Functional/ModelBundle/Repository/NodeRepositoryTest.php [hunk 1]
```diff
@@ -588,4 +588,50 @@ public function provideNodeIdAndLanguageForPublishedFlag()
...
+     * @param string $language
+     *
+     * @dataProvider provideFindByPathCurrentlyPublished
+     */
+    public function testFindByPathCurrentlyPublished($path, $siteId, $expectedCount)
+    {
+        $this->assertCount($expectedCount, $this->repository->findByPathCurrentlyPublished($path, $siteId));
+    }
+
+
```
Comment (alavieille) [inline src/OpenOrchestra/Tests/Functional/ModelBundle/Repository/NodeRepositoryTest.php]: remove empty line

File: src/OpenOrchestra/Tests/Functional/ModelBundle/Repository/NodeRepositoryTest.php [hunk 2]
```diff
@@ -588,4 +588,50 @@ public function provideNodeIdAndLanguageForPublishedFlag()
...
+     * @param string $language
+     *
+     * @dataProvider provideFindLastVersionByTypeCurrentlyPublished
+     */
+    public function testFindLastVersionByTypeCurrentlyPublished($siteId, $expectedCount)
+    {
+        $this->assertCount($expectedCount, $this->repository->findLastVersionByTypeCurrentlyPublished($siteId));
+    }
+
+
```
Comment (alavieille) [inline src/OpenOrchestra/Tests/Functional/ModelBundle/Repository/NodeRepositoryTest.php]: remove this empty line

File: src/OpenOrchestra/Tests/Functional/ModelBundle/Repository/NodeRepositoryTest.php [hunk 3]
```diff
@@ -588,4 +588,50 @@ public function provideNodeIdAndLanguageForPublishedFlag()
...
+    public function provideFindLastVersionByTypeCurrentlyPublished()
+    {
+        return array(
+            array("1", 0),
+            array("2", 5),
+        );
+    }
+
+    /**
+     * @param string $nodeId
```
Comment (alavieille) [inline src/OpenOrchestra/Tests/Functional/ModelBundle/Repository/NodeRepositoryTest.php]: phpdoc

File: src/OpenOrchestra/Tests/Functional/ModelBundle/Repository/NodeRepositoryTest.php [hunk 4]
```diff
@@ -588,4 +588,50 @@ public function provideNodeIdAndLanguageForPublishedFlag()
             'community in en' => array('fixture_page_community', 'en'),
         );
     }
+
+    /**
+     * @param string $nodeId
```
Comment (alavieille) [inline src/OpenOrchestra/Tests/Functional/ModelBundle/Repository/NodeRepositoryTest.php]: php doc

**Comment by itkg-nanne** [comment]
done @alavieille

**Comment by alavieille** [comment]
:+1:

**Pull request merged**